### PR TITLE
Optimization performance for apple device

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: packages/core/src/mesh/SkinnedMeshRenderer.ts [hunk 1]
```diff
@@ -86,6 +89,15 @@ export class SkinnedMeshRenderer extends MeshRenderer {
     super(entity);
     this._mat = new Matrix();
     this._skin = null;
+
+    const rhi = this.entity.engine._hardwareRenderer;
+    let maxVertexUniformVectors = rhi.renderStates.getParameter(rhi.gl.MAX_VERTEX_UNIFORM_VECTORS);
+    if (SystemInfo._isAppleDevice() && !SystemInfo._isChrome()) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/mesh/SkinnedMeshRenderer.ts:95]: What about other browsers for ios

**Review by zhuxudong — APPROVED** [review]
+1

File: packages/rhi-webgl/src/WebGLRenderer.ts [hunk 2]
```diff
@@ -146,6 +151,11 @@ export class WebGLRenderer implements IHardwareRenderer {
     // Make sure the active texture in gl context is on default, because gl context may be used in other webgl renderer.
     gl.activeTexture(gl.TEXTURE0);
 
+    var debugRenderInfo = gl.getExtension("WEBGL_debug_renderer_info");
```
Comment (gz65555) [inline packages/rhi-webgl/src/WebGLRenderer.ts:154]: use `const`

File: packages/core/src/mesh/SkinnedMeshRenderer.ts [hunk 3]
```diff
@@ -86,6 +88,15 @@ export class SkinnedMeshRenderer extends MeshRenderer {
     super(entity);
     this._mat = new Matrix();
     this._skin = null;
+
+    const rhi = this.entity.engine._hardwareRenderer;
+    let maxVertexUniformVectors = rhi.renderStates.getParameter(rhi.gl.MAX_VERTEX_UNIFORM_VECTORS);
+    if (rhi.renderer === "Apple GPU") {
+      // When uniform is large than 256, the skeleton matrix array access in shader very slow in safari and jsc.
```
Comment (gz65555) [inline packages/core/src/mesh/SkinnedMeshRenderer.ts:95]: Is the problem related to jsc?

File: packages/core/src/mesh/SkinnedMeshRenderer.ts [hunk 4]
```diff
@@ -86,6 +88,15 @@ export class SkinnedMeshRenderer extends MeshRenderer {
...
+      // This may be a apple bug, chrome and Firefox is OK!
```
Comment (gz65555) [inline packages/core/src/mesh/SkinnedMeshRenderer.ts:96]: `chrome` should be up case.

File: packages/rhi-webgl/src/WebGLRenderer.ts [hunk 5]
```diff
@@ -79,6 +80,10 @@ export class WebGLRenderer implements IHardwareRenderer {
     return this._isWebGL2;
   }
 
+  get renderer(): String {
```
Comment (gz65555) [inline packages/rhi-webgl/src/WebGLRenderer.ts:83]: should be `string`

**Pull request merged**